ledger/src/blockstore.rs

```diff
@@ -1183,10 +1199,16 @@ impl Blockstore {
             self.get_index_meta_entry(slot, index_working_set, index_meta_time_us);
 
         let index_meta = &mut index_meta_working_set_entry.index;
+        let erasure_set = shred.erasure_set();
+
+        if let HashMapEntry::Vacant(entry) = merkle_root_metas.entry(erasure_set) {
+            if let Some(meta) = self.merkle_root_meta(erasure_set).unwrap() {
+                entry.insert(meta);
+            }
+        }
```

Comment:
I don't understand why merkle-root meta is read from the blockstore.
shouldn't it be obtained from the current shred?

Reply:
the idea here is to load the merkle entry we already have in blockstore into the in memory map for comparison in the future pr